pulsar-broker/src/test/java/org/apache/pulsar/broker/intercept/BrokerInterceptorTest.java

```diff
@@ -179,6 +180,23 @@ public void testBeforeSendMessage() throws PulsarClientException {
         assertEquals(((CounterBrokerInterceptor)listener).getMessageDispatchCount(),1);
     }
 
+    @Test
+    public void testInterceptAck() throws Exception {
+        final String topic = "test-intercept-ack" + UUID.randomUUID();
+        BrokerInterceptor interceptor = pulsar.getBrokerInterceptor();
+        Assert.assertTrue(interceptor instanceof CounterBrokerInterceptor);
+
+        try (Producer<String> producer = pulsarClient.newProducer(Schema.STRING).topic(topic).create();
+             Consumer<String> consumer = pulsarClient.newConsumer(Schema.STRING).topic(topic)
+                     .subscriptionName("test-sub").subscribe()) {
+            producer.send("test intercept ack message");
+            Message<String> message = consumer.receive();
+            consumer.acknowledge(message);
+        }
+
+        Assert.assertEquals(((CounterBrokerInterceptor) interceptor).getHandleAckCount(), 1);
```

Comment:
Need to add `Awaitility.await()` here, for Pulsar consumers, it will not send out the acknowledge command immediately, without some retry policies, here should be a flaky test.

Reply:
@codelipenghui  I've read org.apache.pulsar.client.api.Consumer before, when consumer is closing, it will invoke AcknowledgmentsGroupingTracker#flush to flush all pending acks to broker. I assert the value handleAckCount after consumer is closed, it seems ok

Reply:
Ok, I see. Thanks for the explanation.